test/cluster/mv/test_mv_topology_change.py

```diff
@@ -201,3 +204,47 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
         # will be held for long time until the write timeouts.
         # Otherwise, it is expected to complete in short time.
         await manager.remove_node(servers[0].server_id, servers[-1].server_id, timeout=180)
+
+async def test_mv_pairing_during_replace(manager: ManagerClient):
+    servers = await manager.servers_add(3, property_file=[
+        {"dc": "dc1", "rack": "r1"},
+        {"dc": "dc1", "rack": "r1"},
+        {"dc": "dc1", "rack": "r2"}
+    ])
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 2};")
+    await cql.run_async("CREATE TABLE ks.t (pk int PRIMARY KEY, v int)")
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.t_view AS SELECT pk, v FROM ks.t WHERE v IS NOT NULL PRIMARY KEY (v, pk)")
+
+    stop_event = asyncio.Event()
+    async def do_writes() -> int:
```

Comment:
I don't want to block this PR from being merged, but in the future we should avoid non-deterministic tests. I think you could write a deterministic test for the bug, e.g. shut down the node, make sure that the tablet load balancer wakes up and eventually finishes its work, try to write to the base table and observe that the failed pairing metric does not increase.

Not sure about the replace as we need to check various stages where this could fail. I think we had some tests like that for the topology coordinator which introduce failures at different steps of the operation. Perhaps we could pause the replace operation at different steps, write and check if everything went OK.